src/halmos/calldata.py

```diff
@@ -210,3 +247,71 @@ def head_size(x):
         static = len(tails) == 0
 
         return EncodingResult(heads + tails, total_size, static)
+
+
+@dataclass(frozen=True)
+class FunctionInfo:
+    name: str | None = None
+    sig: str | None = None
+    selector: str | None = None
+
+
+def str_abi(item: dict) -> str:
```

Comment:
not super clear what the contract for this is given the type info and lack of comments

Reply:
added comment